replica/table.cc

```diff
@@ -3223,6 +3223,7 @@ future<> table::cleanup_tablet(database& db, db::system_keyspace& sys_ks, locato
         // the actually flushed/deleted sstables, like discard_sstable() does, so that the mutations
         // cleaned from sstables are exactly the same as the ones cleaned from commitlog.
         co_await sys_ks.save_range_cleanup_record(schema()->id(), sg->token_range(), _highest_rp);
+        co_await sys_ks.drop_old_range_cleanup_records(db.commitlog()->min_position());
```

Comment:
The comment about being less that ideal belongs here too, since it's valuable for more than explaining the patch.

Reply:
Done.